### PR TITLE
Added precompress script

### PR DESCRIPTION
**Pull request by VictorLeach96**
This adds the ability to inject a script before compression.
Can be used for code signing the binaries.

Test needs to be added still.

### CODE REVIEW

File: cmakebuild.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  log.verbose('executing preinstall')
+  util.exec(opts.preinstall, function (err) {
+    if (err) return cb(err)
+    run()
+  })
+
+  function run () {
+    var cmakeJsPath = path.join(
+      __dirname,
+      '../',
```
Comment (ralphtheninja) [inline cmakebuild.js]: `'../.bin'`

File: cmakebuild.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
+var util = require('./util')
+var path = require('path')
+var spawn = require('child_process').spawn
+
+function runGyp (opts, target, cb) {
```
Comment (ralphtheninja) [inline cmakebuild.js]: Maybe rename to `runCmake` instead

File: README.md [hunk 3]
```diff
@@ -121,6 +121,7 @@ prebuild [options]
   --upload      -u  [gh-token]  (upload prebuilds to github)
   --upload-all  -u  [gh-token]  (upload all files from ./prebuilds folder to github)
   --preinstall  -i  script      (run this script before prebuilding)
+  --precompress -c  script      (run this script before compression)
```
Comment (ralphtheninja) [inline README.md]: I'm :-1: on introducing the precompress script in the same pull request as cmake-js.

File: build.js [hunk 4]
```diff
@@ -1,18 +1,28 @@
...
   var log = opts.log || noop
   var release = releaseFolder(opts, version)
 
-  log.verbose('starting node-gyp process')
-  gypbuild(opts, version, function (err) {
-    if (err) return cb(err)
-    log.verbose('done node-gyp\'ing')
-    collectArtifacts(release, opts, cb)
-  })
+  if (opts.backend === 'cmake-js') {
```
Comment (ralphtheninja) [inline build.js]: We can refactor this into one code block, e.g. something like

```js
var build = opts.backend === 'cmake-js' ? cmakebuild : gypbuild
log.verbose('starting build process', opts.backend)
build(opts, version, function (err) {
  if (err) return cb(err)
  collectArtifacts(release, opts, cb)
})
```

File: README.md (hunk 3 above)
Comment (ralphtheninja) [reply README.md]: I just noticed it was two PRs, my bad.

Comment (VictorLeach96) [reply README.md]: Yeh sorry it's a bit messy as this PR is a branch off the cmake one

**Comment by VictorLeach96** [comment]
Rebase had too many conflicts so I created a new branch, see #218

**Pull request closed without merging**